records/validation.go

```diff
@@ -0,0 +1,49 @@
+package records
+
+import (
+	"fmt"
+	"net"
+)
+
+// validateMasters checks that each master in the list is a properly formatted host:ip pair.
+// duplicate masters in the list are not allowed.
+// returns nil if the masters list if empty, or else all masters in the list are valid.
+func validateMasters(ms []string) error {
+	valid := map[string]struct{}{}
+	for i, m := range ms {
+		h, p, err := net.SplitHostPort(m)
+		if err != nil {
+			return fmt.Errorf("illegal host:port specified for master %q", ms[i])
+		}
+		// normalize ipv6 addresses
+		if ip := net.ParseIP(h); ip != nil {
+			h = ip.String()
+			m = h + "_" + p
```

Comment:
ipv6 addresses use `:` as segment delimiters, so I wanted to use a different delimiter when generating the map keys to avoid confusion. `_` seemed ok because it's not allowed in ipv4 or ipv6 IP's.